sources/packages/backend/apps/api/src/route-controllers/student/student.controller.service.ts

```diff
@@ -64,25 +71,57 @@ export class StudentControllerService {
     groupName: string,
     auditUserId: number,
   ): Promise<FileCreateAPIOutDTO> {
-    const createdFile = await this.fileService.createFile(
-      {
-        fileName: file.originalname,
-        uniqueFileName: uniqueFileName,
-        groupName: groupName,
-        mimeType: file.mimetype,
-        fileContent: file.buffer,
-      },
-      studentId,
-      auditUserId,
-    );
-
-    return {
-      fileName: createdFile.fileName,
-      uniqueFileName: createdFile.uniqueFileName,
-      url: `student/files/${createdFile.uniqueFileName}`,
-      size: createdFile.fileContent.length,
-      mimetype: createdFile.mimeType,
-    };
+    let uploadStatus: number;
+    try {
+      this.logger.log(`Uploading file ${file.originalname} to S3 storage.`);
+      uploadStatus = await this.objectStorageService.putObject({
+        key: uniqueFileName,
+        contentType: file.mimetype,
+        body: file.buffer,
+      });
+    } catch (error: unknown) {
+      this.logger.error(parseJSONError(error));
+    }
+    if (uploadStatus === HttpStatus.OK) {
+      const createdFile = await this.fileService.createFile(
+        {
+          fileName: file.originalname,
+          uniqueFileName: uniqueFileName,
+          groupName: groupName,
+        },
+        studentId,
+        auditUserId,
+      );
+      if (createdFile) {
+        this.logger.log(`Uploaded file ${file.originalname} to S3 storage.`);
```

Comment:
The error log is misleading. It should state the file was saved to database.

Reply:
I had put this log here because we can confirm a successful file content upload to S3 only after a successful file save to our db (which can be confirmed only after a successful `createFile`). If the `createFile` fails, we are removing the file from S3. Therefore, I couldn't put this message above.

Reply:
The comment is no longer relevant after the refactor.